src/transformers/utils/generic.py

```diff
@@ -306,11 +308,19 @@ def __init_subclass__(cls) -> None:
         `static_graph=True` with modules that output `ModelOutput` subclasses.
         """
         if is_torch_available():
-            torch_pytree_register_pytree_node(
-                cls,
-                _model_output_flatten,
-                _model_output_unflatten,
-            )
+            if version.parse(get_torch_version()) >= version.parse("2.2"):
+                _torch_pytree.register_pytree_node(
+                    cls,
+                    _model_output_flatten,
+                    partial(_model_output_unflatten, output_type=cls),
+                    serialized_type_name=f"{cls.__module__}.{cls.__name__}",
+                )
+            else:
+                _torch_pytree._register_pytree_node(
+                    cls,
+                    _model_output_flatten,
+                    partial(_model_output_unflatten, output_type=cls),
```

Comment:
My only left (minor) question is: would this introduce any backward compatibility issue (for a user using torch < 2.2 but before and after this PR). 

I think it is `NO` as this is really only in the scope of a python runtime: i.e. there is no saving using code before this PR and loading using code of this PR. (as saving is not working anyway before this PR or torch < 2.2).

So there shouldn't be any compatibility issue. Do you have any comment regarding this?